src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,225 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Duke {
+    public static final String TEXT_SPACER = "    ";
+    public static final java.nio.file.Path path = java.nio.file.Paths.get(System.getProperty("user.dir"));
+    public static final ArrayList<Task> userLists = new ArrayList<>();
+
+    public static void main(String[] args) {
+        printWelcome();
+        printLineSpacer();
+        System.out.println(TEXT_SPACER + "Hey, what's up!\n" + TEXT_SPACER + "What can I help you with today?");
+        printLineSpacer();
+        Scanner scanner = new Scanner(System.in);
+        String userLineInput;
+        int numOfTasks = 0;
+
+        numOfTasks = getFile(userLists, numOfTasks);
+
+        do {
+            userLineInput = scanner.nextLine();
+            String[] splitInputs = userLineInput.split(" ");
+            String userCommand = splitInputs[0];
+            printLineSpacer();
+            if (!userCommand.equals(("bye"))) {
+                try {
+                    switch (userCommand) {
+                    case "list":
+                        printList(userLists, numOfTasks);
+                        break;
+                    case "done":
+                        try {
+                            completeTask(userLists, splitInputs[1]);
+                        } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
+                            System.out.println("Please choose a valid list entry.");
+                        }
```

Comment:
How about putting the `try-catch` statement in the `completeTask` method? This can enhance readability and ensure [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) applies.  You could do the same for other cases.